api/test/service/store.test.js

```diff
@@ -97,10 +98,49 @@ test('store/add returns signed url for uploading', async (t) => {
     with: spaceDid,
     link,
     headers: {
+      'content-length': size,
       'x-amz-checksum-sha256': base64pad.baseEncode(link.multihash.digest)
     }
   })
-  t.is(storeAdd.url && new URL(storeAdd.url).pathname, `/${link}/${link}.car`)
+  const url = storeAdd.url && new URL(storeAdd.url)
+  if (!url) {
+    throw new Error('Expected presigned url in response')
+  }
+
+  const signedHeaders = url.searchParams.get('X-Amz-SignedHeaders')
+  t.is(signedHeaders, 'content-length;host;x-amz-checksum-sha256', 'content-length and checksum must be part of the signature')
+
+  const key = url?.pathname.slice(11) // minio puts bucket name at start of path
+  t.is(key, `/${link}/${link}.car`)
+
+  const goodPut = await fetch(url, {
+    method: 'PUT',
+    mode: 'cors',
+    body: data,
+    headers: storeAdd.headers
+  })
+  t.is(goodPut.status, 200, await goodPut.text())
+
+  const unexpectedLength = await fetch(url, {
+    method: 'PUT',
+    mode: 'cors',
+    body: new Uint8Array([11, 22, 34, 44, 55, 1]), // one byte more
+    headers: {
+      'content-length': '6',
+      'x-amz-checksum-sha256': storeAdd.headers['x-amz-checksum-sha256']
+    }
+  })
+  // expect signature to not match
+  t.is(unexpectedLength.status, 403, await unexpectedLength.text())
+
+  // TODO: why is minio not respecting the checksum header?
```

Comment:
Can we crease an issue there?

Reply:
https://github.com/web3-storage/upload-api/issues/72